autopush/router/fcm.py

```diff
@@ -32,8 +31,87 @@ def __init__(self, ap_settings, router_conf):
         self.auth = router_conf.get("auth")
         self.metrics = ap_settings.metrics
         self._base_tags = []
+        self.creds = {}
+        self.regid = ""
+        self.reasonTable = {
+            "MissingRegistration": {
+                "msg": ("'to' or 'registration_id' is blank or"
+                        " invalid: {regid}"),
+                "err": 500,
+                "errno": 1,
+            },
+            "InvalidRegistration": {
+                "msg": "registration_id is invalid: {regid}",
+                "err": 410,
+                "errno": 105,
+                "crit": True,
```

Comment:
I'll agree about InvalidRegistration & DeviceMessageRateExceeded (since I missed those when I was cleaning up) but the other ones imply something seriously wrong with either our code or that the FCM endpoint has changed. I'm willing to say those are critical.